### PR TITLE
[DRIVR-4381] towards api tokens only

### PR DESCRIPTION
**Pull request by hemarkus**
**Checklist**

- [ ] Commits are reasonably small, easy to follow, and only concern one topic or task
- [ ] Commit messages are descriptive and include more details in the body if necessary
- [ ] Acceptance criteria including constraints are present in the ticket or linked
- [ ] There is no communication or clarifications that has not been added to the ticket
- [ ] No further unit tests need to be added or updated
- [ ] No further component test cases need to be adjusted
- [ ] The specification and documentation are up-to-date with the implementation
- [ ] Link to Pull Request is added to the ticket (Add Tickets to title)

### CODE REVIEW

**Review by clstb — APPROVED** [review]

File: api/api.go [hunk 1]
```diff
@@ -107,7 +100,10 @@ func (d *DrivrAPI) FetchCertificateAuthority(ctx context.Context, issuer string)
 		logrus.WithError(err).Error("Failed to decode CA certificate")
 		return nil, err
 	}
-	return ca, nil
+
+	derBlock, _ := pem.Decode(ca)
```
Comment (milj) [inline api/api.go:104]: What is the meaning of "der"?

**Review by milj — APPROVED** [review]

File: api/api.go (hunk 1 above)
Comment (hemarkus) [reply api/api.go:104]: DER (Distinguished Encoding Rules)
That is the format the certificate is in at that point. Actually PEM is the base64 encoded certificate in DER (which is binary) plus a header and a footer.

**Comment by hemarkus** [comment]
Review of the documentation revealed it is not a good idea to allow passing the api-key via a flag. Changed to use env variable or input at runtime.

File: api/api.go (hunk 1 above)
Comment (milj) [reply api/api.go:104]: Deshalb I would add a comment about DER :)

**Review by clstb — APPROVED** [review]

File: cmd/drivr-certificate-client/shared.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+		Usage:   "Issuer of the certificate",
+	}
+)
+
+const drivrAPIKeyEnv = "DRIVR_API_KEY"
+
+var apikey string
+
+func getAPIKey() string {
+	// read the API key from the environment
```
Comment (milj) [inline cmd/drivr-certificate-client/shared.go:54]: This happens a few lines below, can this comment be moved down?

File: cmd/drivr-certificate-client/shared.go [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+	keyBytes, err := term.ReadPassword(int(syscall.Stdin))
+	if err != nil {
+		logrus.WithError(err).Error("Failed to read API key")
+		return ""
+	}
+	apikey = string(keyBytes)
+	return apikey
+}
+
+func combinedCheckFuncs(checks ...func(*cli.Context) error) func(*cli.Context) error {
```
Comment (milj) [inline cmd/drivr-certificate-client/shared.go:74]: Elegant.

**Review by milj — APPROVED** [review]

**Pull request merged**